### PR TITLE
[ITBL-1163] in app notifications mobile ui

### PR DESCRIPTION
**Pull request by davidtruong**
In progress work for in app notification. Will finalize once we get the backend hooked up.

### CODE REVIEW

File: Iterable-iOS-SDK/IterableAlertView.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
+//
+//  IterableAlertView.h
+//
+//  Implementation based of of NYAlert created by Nealon Young
+//  Copyright (c) 2015 Nealon Young. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+typedef NS_ENUM(NSInteger, IterableAlertViewButtonType) {
```
Comment (dontgitit) [inline Iterable-iOS-SDK/IterableAlertView.h]: NSInteger vs NSUInteger in the other enum?

Comment (dontgitit) [inline Iterable-iOS-SDK/IterableAlertView.h]: i think you can add docstrings for these somehow

File: Iterable-iOS-SDK/IterableFullScreenViewController.m [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+@interface IterableFullScreenViewController ()
+@property (nonatomic, strong) UIImageView* ImageView;
+@property (nonatomic) NSDictionary *actionButtons;
+
+@end
+
+@implementation IterableFullScreenViewController
+
+CGFloat imageWidth =0;
```
Comment (dontgitit) [inline Iterable-iOS-SDK/IterableFullScreenViewController.m]: weird spacing

File: Iterable-iOS-SDK/IterableInAppManager.m [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    IterableInAppBaseViewController *baseNotification;
+    
+    NSString* type = [payload objectForKey:ITERABLE_IN_APP_TYPE];
+    if ([type isEqual:ITERABLE_IN_APP_TYPE_FULL]){
+        baseNotification = [[IterableFullScreenViewController alloc] init];
+    } else {
+        baseNotification = [[IterableAlertViewController alloc] initWithNibName:nil bundle:nil];
+    }
+    
+    //TODO: add in view tracking
```
Comment (dontgitit) [inline Iterable-iOS-SDK/IterableInAppManager.m]: TODO?

File: Iterable-iOS-SDK/IterableConstants.h [hunk 4]
```diff
@@ -0,0 +1,34 @@
+//
+//  IterableConstants.h
+//  Iterable-iOS-SDK
+//
+//  Created by David Truong on 9/9/16.
```
Comment (jontsai) [inline Iterable-iOS-SDK/IterableConstants.h:5]: my personal pref is to remove comments like this

**Comment by dontgitit** [comment]
adds some docs for the public apis, otherwise lftm

File: Iterable-iOS-SDK/IterableAPI.h [hunk 5]
```diff
@@ -52,6 +54,12 @@ typedef NS_ENUM(NSInteger, PushServicePlatform) {
 @property(nonatomic, readonly, copy) NSString *email;
```
Comment (hjz) [inline Iterable-iOS-SDK/IterableAPI.h:54]: can we add a comment, that the email is the preferred key

File: Iterable-iOS-SDK/IterableAPI.h [hunk 6]
```diff
@@ -63,27 +71,20 @@ typedef NS_ENUM(NSInteger, PushServicePlatform) {
...
  
  @param apiKey          your Iterable apiKey
- @param email           the email of the user logged in
+ @param userId           the userId of the user logged in
  @param launchOptions   launchOptions from application:didFinishLaunchingWithOptions
  
  @return an instance of IterableAPI
  */
-- (instancetype) initWithApiKey:(NSString *)apiKey andEmail:(NSString *)email launchOptions:(nullable NSDictionary *)launchOptions;
++ (IterableAPI *) sharedInstanceWithApiKey:(NSString *)apiKey andUserId:(NSString *)userId launchOptions:(nullable NSDictionary *)launchOptions;
```
Comment (hjz) [inline Iterable-iOS-SDK/IterableAPI.h:141]: Is the email option removed?

File: Iterable-iOS-SDK/IterableAPI.m [hunk 7]
```diff
@@ -86,6 +87,32 @@ - (NSURL *)getUrlForAction:(NSString *)action
...
+- (NSURL *)getUrlForGetAction:(NSString *)action withArgs:(NSDictionary *)args
+{
+    NSString *urlCombined = [NSString stringWithFormat:@"%@%@?api_key=%@", endpoint, action, self.apiKey];
+    //updated this to take in a dictionary are parse values
+    
+    
+    for (NSString* paramKey in args) {
+        NSString* paramValue = args[paramKey];
+        
+        NSString *params = [NSString stringWithFormat:@"&&%@=%@", paramKey, paramValue];
```
Comment (andrewboni) [inline Iterable-iOS-SDK/IterableAPI.m]: single `&`

File: Iterable-iOS-SDK/IterableAPI.h (hunk 6 above)
Comment (davidtruong) [reply Iterable-iOS-SDK/IterableAPI.h:141]: It has its own function with a separate call signature. It is still there as: + (IterableAPI *) sharedInstanceWithApiKey:(NSString *)apiKey andEmail:(NSString *)email

File: Iterable-iOS-SDK/IterableAPI.h (hunk 5 above)
Comment (davidtruong) [reply Iterable-iOS-SDK/IterableAPI.h:54]: Added the comment in the function and in the README docs.

**Review by davidtruong — COMMENTED** [review]
Made changes

File: Iterable-iOS-SDK/IterableAPI.m (hunk 7 above)
Comment (davidtruong) [reply Iterable-iOS-SDK/IterableAPI.m]: done

**Pull request merged**